src/components/application_manager/test/commands/mobile/put_file_test.cc

```diff
@@ -0,0 +1,113 @@
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <stdint.h>
+#include <string>
+
+#include "gtest/gtest.h"
+
+#include "application_manager/commands/commands_test.h"
+
+#include "mobile/put_file_response.h"
+#include "mobile/put_file_request.h"
+
+#include "utils/make_shared.h"
+#include "smart_objects/smart_object.h"
+
+#include "application_manager/mock_application.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+
+using namespace application_manager;
+
+using ::testing::Return;
+using ::testing::_;
+
+class PutFileResponceTest : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  void SetUp() OVERRIDE {
+    message_ = utils::MakeShared<SmartObject>(::smart_objects::SmartType_Map);
+    (*message_)[strings::msg_params] =
+        ::smart_objects::SmartObject(::smart_objects::SmartType_Map);
+
+    command_sptr_ =
+        CreateCommand<application_manager::commands::PutFileResponse>(
+            CommandsTest::kDefaultTimeout_, message_);
+  }
+
+  MessageSharedPtr message_;
+  utils::SharedPtr<commands::PutFileResponse> command_sptr_;
+};
+
+TEST_F(PutFileResponceTest, Run_InvalidApp_ApplicationNotRegisteredResponce) {
+  const int32_t kConnectionKey = 1;
+  ::smart_objects::SmartObject& message_ref = *message_;
+
+  message_ref[strings::params][strings::connection_key] = kConnectionKey;
+
+  utils::SharedPtr<Application> null_application_sptr;
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(null_application_sptr));
+  EXPECT_CALL(
+      mock_app_manager_,
+      SendMessageToMobile(
+          MobileResultCodeIs(mobile_api::Result::APPLICATION_NOT_REGISTERED),
+          _)).Times(1);
+  command_sptr_->Run();
+}
+
+TEST_F(PutFileResponceTest, Run_ApplicationRegistered_Success) {
+  const int32_t kConnectionKey = 1;
```

Comment:
@Kozoriz, you define `kConnectionKey` in both tests. Please move it to nameless namespace.